### DISCUSSION TITLE
A conjugate heat transfer problem about TH module.

### DISCUSSION THREAD

**Discussion by Raven-pro** [idaholab/moose#22843] [Q&A General] [answered]
Hello developers!

I am using the TH module to establish a conjugate heat transfer system where there is a 1D flow channel at the center, and there are two heat structures on each side.

The structure of the heat structure is the fuel as a heat source inside with cladding outside.

So the process is, the heat source generates heat, and the heat is conducted through the cladding, the flow takes away heat as its temperature increases.

When the simulation reaches a steady state, the temperature distribution should be, the heat source having the highest temperature, and near the wall, the temperature is lower.

However, the result is different from what it is supposed to be. I visualized it in the ParaView as below.
<img width="694" alt="截屏2022-12-01 20 50 30" src="https://user-images.githubusercontent.com/68334741/205057504-261af0c8-ff1a-4020-935a-3effa5faa3cf.png">
So, for each heat structure, the heat distribution is identical: the highest temperature on the left and the lowest on the right
I have also attached my input file below. Can anybody help me with dealing with such a problem?
```
T_in = 533.15            # K
# m_dot_in = 1.3565      # kg/s
m_dot_in = 0.96805       # kg/s
press = 15.5e6           # Pa

# plate channels parameters (x,y,z)
plate_height = 0.75      # m
plate_width = 0.0762     # m
plate_thick = 0.00038    # m

# core parameters
core_n_elems = 100
core_gap = 0.00228       # m

tot_power = 2e9          # W/m3

[GlobalParams]
    initial_p = ${press}
    initial_vel = 7
    initial_T = ${T_in}
    rdg_slope_reconstruction = full
    closures = simple_closures
    fp = water
[]

# [Functions]
#     [power_func]
#         type = ParsedFunction
#         value = '1.570796326794897 * sin(z / 3.6576 * pi)'
#     []
# []

[Functions]
    [HeatFunction]
      type = ParsedFunction
      value = ${tot_power}
    []
[]

# [AuxVariables]
#     [power_density]
#       family = MONOMIAL
#       order = CONSTANT
#       block = 'core_hs1:FUEL1 core_hs2:FUEL2'
#     []
# []

# [AuxKernels]
#     [mock_power_aux]
#       type = FunctionAux
#       variable = power_density
#       function = HeatFunction
#       block = 'core_hs1:FUEL1 core_hs2:FUEL2'
#     []
# []

[Modules]
    [FluidProperties]
        [water]
            type = StiffenedGasFluidProperties
            gamma = 2.35
            cv = 1816.0
            q = -1.167e6
            p_inf = 1.0e9
            q_prime = 0
        []
    []
[]

[SolidProperties]
    [cladding]
        type = ThermalFunctionSolidProperties
        rho = 2700.0
        cp = 897
        k = '131.25+t/8.33e2'
    []

    [fuel]
        type = ThermalFunctionSolidProperties
        rho = 6030.0
        cp = '(0.892+0.00046*t-0.71*(0.749+0.00038*t))*1e3'
        k = 50.7
    []
[]


[Closures]
    [simple_closures]
        type = Closures1PhaseSimple
    []
[]
########################Structure Material Prop#################
[Materials]
    [fuel-mat]
      type = ADConstantDensityThermalSolidPropertiesMaterial
      block = 'core_hs1:FUEL1 core_hs2:FUEL2'
      sp = fuel
      temperature = T_solid
      T_ref = 300
    []
  
    # [gap-mat]
    #   type = ADGenericConstantMaterial
    #   block = 'core_hs:GAP1 core_hs:GAP2'
    #   prop_names = 'thermal_conductivity specific_heat density'
    #   prop_values = '1.084498 1.0 1.0'
    # []
  
    [clad-mat]
      type = ADConstantDensityThermalSolidPropertiesMaterial
      block = 'core_hs1:CLAD1 core_hs2:CLAD2'
      sp = cladding
      temperature = T_solid
      T_ref = 300
    []
[]

[Components]
########################Single Phase Flow#######################

############################channel 1###########################
    [inlet1]
        type = InletMassFlowRateTemperature1Phase
        input = 'core_chan1:in'
        m_dot = ${m_dot_in}
        T = ${T_in}
    []

    [core_chan1]
        type = FlowChannel1Phase
        position = '0 0 0'
        orientation = '0 0 1'
        length = ${plate_height}
        n_elems = ${core_n_elems}
        A = ${fparse plate_width * core_gap}
        D_h = ${fparse 2 * plate_width * core_gap/(plate_width + core_gap)}
        f = 0.16
    []

    [outlet1]
        type = Outlet1Phase
        input = 'core_chan1:out'
        p = ${press}
    []
########################Heat Structure##########################

############################Core HS1############################
    [total_power1]
        type = TotalPower
        power = ${fparse tot_power * plate_height * plate_thick * plate_width}
    []
    [core_hs1]
        type = HeatStructureCylindrical
        position = '0 0 0'           
        orientation = '0 0 1'
        length = ${plate_height}
        n_elems = ${core_n_elems}

        names = 'CLAD1 FUEL1'       # pay attention to block
        widths = '0.00038 0.00038'
        # materials = 'steel'
        n_part_elems = '10 10'      # radial elements
        initial_T = 564.15
    []

    [core_heating1]
        type = HeatSourceFromTotalPower
        hs = core_hs1
        regions = 'FUEL1'
        power = total_power1
        # power_shape_function = power_func
    []

    # [core_heating1]
    #     type = HeatSourceFromPowerDensity
    #     hs = core_hs1
    #     regions = 'FUEL1'
    #     # power_density = power_density
    #     power = total_power
    # []

    [core_ht1]
        type = HeatTransferFromHeatStructure1Phase
        flow_channel = core_chan1
        hs = core_hs1
        hs_side = outer
        P_hf = ${plate_width}
        Hw = 50000
    []

############################Core HS2############################
    [total_power2]
        type = TotalPower
        power = ${fparse tot_power * plate_height * plate_thick * plate_width}
    []
    [core_hs2]
        type = HeatStructureCylindrical
        position = '0 -0.00076 0'           
        orientation = '0 0 1'
        length = ${plate_height}
        n_elems = ${core_n_elems}

        names = 'CLAD2 FUEL2'       # pay attention to block
        widths = '0.00038 0.00038'
        # materials = 'steel'
        n_part_elems = '10 10'      # radial elements
        initial_T = 564.15
    []

    [core_heating2]
        type = HeatSourceFromTotalPower
        hs = core_hs2
        regions = 'FUEL2'
        power = total_power2
        # power_shape_function = power_func
    []

    # [core_heating2]
    #     type = HeatSourceFromPowerDensity
    #     hs = core_hs2
    #     regions = 'FUEL2'
    #     # power_density = power_density
    #     power = total_power
    # []

    [core_ht2]
        type = HeatTransferFromHeatStructure1Phase
        flow_channel = core_chan1
        hs = core_hs2
        hs_side = outer
        P_hf = ${plate_width}
        Hw = 50000
    []
########################End#####################################
[]

[Executioner]
    type = Transient
    start_time = 0
    end_time = 100000
    dt = 10000
  
    line_search = basic
    solve_type = PJFNK
  
    nl_rel_tol = 1e-5
    nl_abs_tol = 1e-5
    nl_max_its = 5
[]
  
[Outputs]
    exodus = true
  
    [console]
      type = Console
      max_rows = 1
      outlier_variable_norms = false
    []
    print_linear_residuals = false
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

You need to pass different sides in hs_side for the two heat transfer components I think.
From the docs https://mooseframework.inl.gov/modules/thermal_hydraulics/source/components/HeatTransferFromHeatStructure1Phase.html
You have these options:
END, INNER, OUTER, START

I think you want one to be 'inner' and the other to be outer. 

Am I understanding this right that one cylinder is the inner cylinder and the other the outer cylinder?

Guillaume

**Reply by Raven-pro** [reply]
Thanks, I tried to switch `hs_side` to `inner`, but the following warning came out.

<img width="804" alt="截屏2022-12-02 10 36 06" src="https://user-images.githubusercontent.com/68334741/205202228-a4f0f34a-6979-4644-b2c8-28799909f64a.png">

I don't know what `inner` and `outer` mean here. I want the heat to transfer from left to right in one cylinder and from right to left in the other cylinder, as the flow channel is at the center. 

But the result seems that the heat can only transfer from left to right.

**Reply by Raven-pro** [reply]
Am I sitting something wrong in my input file?

**Reply by GiudGiud** [reply]
so the flow is between two cylinders that are NOT concentric? do you have a sketch of what the channel looks like?

**Reply by GiudGiud** [reply]
as to what the meaning of inner and outer are, these structures are cylinders, so it's the inner/outer surfaces of a cylinder that has a hole inside if inner_radius > 0

**Reply by Raven-pro** [reply]
Yes, the cylinders are not concentric, they are parallel with the flow in the middle. The sketch is as below.
![捕获](https://user-images.githubusercontent.com/68334741/205218258-18641946-1ebc-4451-a5fe-19b986a29887.PNG)

Can TH module achieve such conjugate heat transfer?

**Reply by GiudGiud** [reply]
it can

I think something is wrong with your positioning of each structure. They overlap and touch and it seems wrong
- one heat structure is at y = -0.0076 and is 0.0076 * 2 wide so it touches y=0
- the channel is at y=0
- the other heat structure is at y=0 so it overlaps with the channel and it s also 0.0076*2 wide so it overlaps with the other heat structure. 

Guillaume

**Reply by Raven-pro** [reply]
I don't get it, I thought the first heat structure is 0.0076 wide(2 * 0.0038).

Because the channel is at y = 0, and the left edge of the second structure just touches the channel.

The left edge of the first structure is -0.0076, so the right edge of it touches the channel as well.

I don't think there is overlapping. Did I miss something?

**Reply by GiudGiud** [reply]
The plot you showed seems to show the two heat structures are touching, when you want the space for the channel in between?

**Reply by Raven-pro** [reply]
The channel(the blue part) is 1D, and is in the middle of two heat structures(the red parts are fuels, and the orange parts are claddings).

As the channel is 1D, and I thought the heat structures should touch the channel(I don't know if that's true), so the right side of the left hs and the left side of the right hs touch the same flow channel, and it seems they touch each other.

**Reply by GiudGiud** [reply]
I can see that in the sketch. In the exodus output you pasted first, is every component where it should be?

This just looks like an output issue at this point. We only output the RZ part of the cylinder, and the one you want to see in the figure is the symmetric with regards to the Z axis of this RZ figure.

**Reply by Raven-pro** [reply]
Oh！ You are right!
I think I know where the problem is!
Thank you!
